src/api.rs

```diff
@@ -4,6 +4,9 @@ pub enum DriverType {
 }
 
 pub trait Introspectable {
+    // read physical memory
+    fn read_physical(&self, paddr: u64, count: u32) -> Result<Vec<u8>,&str>;
```

Comment:
@tathanhdinh should this return a `Result<&mut [u8], &str>`, or a `Vec` ? I'm a bit lost here.

Reply:
Actually an array sounds better than a Vec, since we know the maximum size in advance, and it shouldn't not grow anyway.